app/common/middleware/locals.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * Legacy query handling
+ */
+const locals = (req, res, next) => {
+  res.locals.legacy = ['1', 'true'].includes(req.query.legacy)
```

Comment:
I'd also consider splitting this refactor out into a separate commit.

Reply:
✅ Done

I've configured Express.js with the standard [`URLSearchParams()`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) too ready for Express.js v5

```mjs
const { query } = req

// Check for query
query.has('example')

// Get query value
query.get(example')

// Set query value
query.set('example', 'value')
```

Reply:
Is that because of the [change of the default query parser to `simple`](https://heynode.com/blog/2020-08/expressjs-5-nodes-latest-web-framework-release/#default-to-simple-query-parser-setting)?

I think the simple query parser would still work for our use case as we're not doing anything fancy with nested params etc. Do you think there's advantages to using `URLSearchParams` over it?

Reply:
Ah I remember that, there was a security vulnerability wasn't there

Primarily because we've now got the WHATWG [URL](https://url.spec.whatwg.org) and [Fetch](https://fetch.spec.whatwg.org) standards so it's all in Node.js too:

### Query to string
```mjs
// 'example1=value1&example2=value2'
req.query.toString()
```

### Query to object
```mjs
// { example1: 'value1', example2: 'value2' }
Object.fromEntries(req.query)
```

### Query via URL object
Where `window.location.search === '?example1=value1&example2=value2'`

```mjs
const url = new URL('http://localhost')

url.search = '?example1=value1&example2=value2'

url.searchParams.get('example1') // 'value1'
url.searchParams.get('example2') // 'value2'
```

Express.js has said they'll also switch in future
https://github.com/expressjs/express/issues/4990#issuecomment-1241976112

Happy to put it back to objects again